src/ess/reduce/nexus/generic_types.py

```diff
@@ -151,8 +152,11 @@ class MonitorData(
 Component = TypeVar('Component', bound=snx.NXobject)
 
 
+class Filename(sciline.Scope[RunType, Path], Path): ...
+
+
 @dataclass
-class Filename(Generic[RunType]):
+class NeXusFileSpec(Generic[RunType]):
     value: FilePath | NeXusFile | NeXusGroup
```

Comment:
@jl-wynen Wondering what you think about this. My latest attempt to have both, Union-type and (generic) type alias, by splitting it into two parts. For the common user there is a helper that forwards `Filename` into `NeXusFileSpec`.

Reply:
Of course it does not solve the `str | Path` issue, unless we split the types and helpers further.

Reply:
I'm not super happy because this adds another node to the graph.

Reply:
As you can guess this is due to lack of better options. Do you have any?

Reply:
We could duplicate the (current 4) providers that turn `NeXusFileSpec` into `NeXusLocationSpec`. Then the extra node is gone. Is that better?

Reply:
> As you can guess this is due to lack of better options. Do you have any?

Drop `Filename` and auto-convert parameters to their key type on insertion. This would mean that
```python
pl[NeXusFileSpec] = 'some/path'
```
produces the expected `NeXusFileSpec` object.

I suggested this in the past but you decided against it in favour of duck-typing. I think that this precludes us from supporting static type checkers without workarounds like you implemented here.

On a side note, what we really want here is a sum-type that works both at runtime and makes mypy happy. I have been trying to create one several times but have been unsuccessful.

Reply:
I agree, we should consider your suggestion again.